src/xb-arena.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2025 Owen Chiaventone <ochiaventone@gmail.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#pragma once
+
+/* Simple arena allocator used to construct nodes during parsing pass
+ * If the nodes and strings on them were individually heap allocated instead, we would churn the
+ * heap with a large number of small and medium sized allocations. This is bad for performance in
+ * a short lived program, but more importantly can trigger serious heap fragmentation if you parse
+ * a large XML in the middle of a long-lived program.
+ * This was implicated in https://gitlab.gnome.org/GNOME/gnome-software/-/issues/941
+ *
+ * We don't know how large the XML is when we start parsing, so the arena may have to grow.
+ * This arena allocator uses a chunking strategy - if we run out, allocate another large block. 
+ */
```

Comment:
Typically the approach in GLib-ish projects is to put doc comments like this in a gtk-doc block near the top of the C file for the utility (so, `xb-arena.c` in this case, below the `#include`s), as that’s where people look for them and that’s the file which is more likely to be read and modified. We’ve found this gives a greater chance of the comment being kept up to date with future modifications.